sympy/solvers/solveset.py

```diff
@@ -572,9 +576,10 @@ def _solveset(f, symbol, domain, _check=False):
         # wrong solutions we are using this technique only if both f and g are
         # finite for a finite input.
         result = Union(*[solver(m, symbol) for m in f.args])
-    elif _is_function_class_equation(TrigonometricFunction, f, symbol) or \
-            _is_function_class_equation(HyperbolicFunction, f, symbol):
+    elif _is_function_class_equation(TrigonometricFunction, f, symbol):
         result = _solve_real_trig(f, symbol)
+    elif _is_function_class_equation(HyperbolicFunction, f, symbol):
+        result = _solve_hyperbolic(f, symbol)
```

Comment:
after #10709 is merged, I think you can just call `_solve_trig(f, symbol, domain)` -- does that work?